From 96b785539d0bf93cdfeb3fe78e793b984bcf0776 Mon Sep 17 00:00:00 2001
From: Mark Rushakoff <mark@strange.love>
Date: Fri, 8 Nov 2024 11:29:19 -0500
Subject: [PATCH 5/6] chore: make root store Close idempotent

We will see about removing this later, but currently is it necessary to
get gcosmos to shut down properly.
---
 store/v2/root/store.go | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/store/v2/root/store.go b/store/v2/root/store.go
index ab22d3f901..92e964c190 100644
--- a/store/v2/root/store.go
+++ b/store/v2/root/store.go
@@ -81,7 +81,10 @@ func New(
 // Close closes the store and resets all internal fields. Note, Close() is NOT
 // idempotent and should only be called once.
 func (s *Store) Close() (err error) {
-	err = errors.Join(err, s.stateCommitment.Close())
+	if s.stateCommitment != nil {
+		err = errors.Join(err, s.stateCommitment.Close())
+	}
+
 	err = errors.Join(err, s.dbCloser.Close())

 	s.stateCommitment = nil
--
2.47.0

